Source/DafnyLanguageServer/Workspace/NotificationPublisher.cs

```diff
@@ -51,12 +64,19 @@ public class NotificationPublisher : INotificationPublisher {
       languageServer.TextDocument.SendNotification(DafnyRequestNames.VerificationSymbolStatus, notification);
     }
 
-    private static FileVerificationStatus? GetFileVerificationStatus(DafnyDocument document) {
-      if (document.ImplementationIdToView == null || document.VerificationTasks == null) {
+    private static FileVerificationStatus? GetFileVerificationStatus(CompilationView document) {
+      if (!document.ImplementationsWereUpdated) {
+        /*
+         CompilationAfterResolution.Snapshot() gets migrated ImplementationViews.
```

Comment:
Could you elaborate a bit? I'm not sure I understand that point. ImplementationViews are not migrated directly?

Reply:
About this? I just want clarification. Also:
```suggestion
         DocumentAfterResolution.Snapshot() gets migrated ImplementationViews.
```

Reply:
The `ImplementationsWereUpdated` boolean is surprising. This comment is trying to argue why it exists. An IdeState that has not been translated yet gets migrated ImplementationViews so that it can publish correct diagnostics, but this would cause the verification symbol status notification to be published as well, since due to the migration it's literal value has changed since last time, even though semantically, nothing has changed and the IDE doesn't need a new notification. This boolean prevents that notification from being emitted.

I think we could try to remove the boolean, but I didn't want to spend time on that now.